src/lib/y2network/widgets/boot_protocol.rb

```diff
@@ -0,0 +1,364 @@
+require "cwm/custom_widget"
+
+Yast.import "DNS"
+Yast.import "Hostname"
+Yast.import "IP"
+Yast.import "Label"
+Yast.import "Netmask"
+Yast.import "NetHwDetection"
+Yast.import "Popup"
+Yast.import "ProductFeatures"
+Yast.import "UI"
+
+module Y2Network
+  module Widgets
+    class BootProtocol < CWM::CustomWidget
+      def initialize(settings)
+        textdomain "network"
+        @settings = settings
+      end
+
+      def contents
+        RadioButtonGroup(
+          Id(:bootproto),
+          VBox(
+            Left(
+              HBox(
+                RadioButton(
+                  Id(:bootproto_none),
+                  Opt(:notify),
+                  _("No Link and IP Setup (Bonding Slaves)")
+                ),
+                HSpacing(1),
+                ReplacePoint(
+                  Id(:bootproto_rp),
+                  CheckBox(Id(:bootproto_ibft), Opt(:notify), _("Use iBFT Values"))
+                )
+              )
+            ),
+            Left(
+              HBox(
+                RadioButton(Id(:bootproto_dynamic), Opt(:notify), _("Dynamic Address")),
+                HSpacing(2),
+                ComboBox(
+                  Id(:bootproto_dyn),
+                  Opt(:notify),
+                  "",
+                  [
+                    Item(Id(:bootproto_dhcp), "DHCP"),
+                    Item(Id(:bootproto_dhcp_auto), "DHCP+Zeroconf"),
+                    Item(Id(:bootproto_auto), "Zeroconf")
+                  ]
+                ),
+                HSpacing(2),
+                ComboBox(
+                  Id(:bootproto_dhcp_mode),
+                  "",
+                  [
+                    Item(Id(:bootproto_dhcp_both), _("DHCP both version 4 and 6")),
+                    Item(Id(:bootproto_dhcp_v4), _("DHCP version 4 only")),
+                    Item(Id(:bootproto_dhcp_v6), _("DHCP version 6 only"))
+                  ]
+                )
+              )
+            ),
+            VBox(
+              # TODO : Stat ... Assigned
```

Comment:
?

Reply:
I have no idea what it means, it was in original code.

Reply:
that was the point .... doesn't make sense to keep comments which cannot be understand ;-)